### DISCUSSION TITLE
Automatic decoding of custom extension objects

### DISCUSSION THREAD

**Discussion by andreapicaro** [gopcua/opcua#441] [Q&A]
Hello, from my understanding the stack doesn't currently allow automatic decoding of custom extension objects and it requires struct to be defined at client build type. Is this still true? If so, is there eventually a roadmap for that functionality?
Thanks

**Comment by magiconair** [comment]
We don't have that on the roadmap but I am happy to take a PR

**Reply by magiconair** [reply]
Here is some code which does the heavy lifting of generating an anonymous Go type at runtime to help with the decoding. We still need to pull the `ua.StructureDefinition` from the server at runtime though.

https://github.com/gopcua/opcua/discussions/691#discussioncomment-7058178

**Reply by May1592** [reply]
@magiconair @kung-foo How we handle null values while decoding a custom extension object ?

![image](https://github.com/gopcua/opcua/assets/41743235/d7371197-357e-4373-8841-4a27a6c014af)

**Reply by magiconair** [reply]
Is this related to the code fragment or a separate question? In the latter case I’d prefer a new topic.
